django/db/models/expressions.py

```diff
@@ -1252,6 +1255,38 @@ def __repr__(self):
         return "{}({})".format(self.__class__.__name__, self.expression)
 
 
+class NegatedExpression(ExpressionWrapper):
+    """
+    Wrapping an expression to negate its output.
+    """
+
+    def __init__(self, expression):
+        super().__init__(expression, output_field=fields.BooleanField())
+
+    def __invert__(self):
+        return copy.copy(self.expression)
```

Comment:
```suggestion
        return self.expression.copy()
```

Reply:
Hmm, when doing this change some unit tests are failing. But that is potentially because the objects passed are usually not expected in this context.

```
======================================================================
ERROR: test_invert (expressions.tests.CombinableTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/david/django/tests/expressions/tests.py", line 2388, in test_invert
    self.assertIsNot(~~c, c)
  File "/Users/david/django/django/db/models/expressions.py", line 1267, in __invert__
    return self.expression.copy()
AttributeError: 'Combinable' object has no attribute 'copy'

======================================================================
ERROR: test_invert_negated_expression (expressions.tests.NegatedExpressionTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/david/django/tests/expressions/tests.py", line 2548, in test_invert_negated_expression
    self.assertEqual(~expr, F("field"))
  File "/Users/david/django/django/db/models/expressions.py", line 1267, in __invert__
    return self.expression.copy()
AttributeError: 'F' object has no attribute 'copy'
```

Reply:
I'd move `test_invert` to the `NegatedExpressionTests` and use `F` instead of `Combinable`. Unfortunately, this won't fix `AttributeError` for `F` expression :disappointed: Maybe we could add `copy()` to the `F` expression :thinking:

Reply:
Pushed some changes 👍